src/entity/object/AreaEffectCloud.php

```diff
@@ -0,0 +1,384 @@
+		$this->networkPropertiesDirty = true;
+	}
+
+	protected function entityBaseTick(int $tickDiff = 1) : bool{
+		$hasUpdate = parent::entityBaseTick($tickDiff);
+
+		$this->age += $tickDiff;
+		$radius = $this->radius + ($this->radiusPerTick * $tickDiff);
+		if($radius < 0.75){
+			$this->flagForDespawn();
+			return true;
+		}
+		$this->setRadius($radius);
+		if($this->age >= $this->waiting){
+			if($this->age > $this->duration){
+				$this->flagForDespawn();
+				return true;
+			}
+			foreach($this->victims as $entityId => $expiration){
+				if($this->age >= $expiration){
+					unset($this->victims[$entityId]);
+				}
+			}
+			foreach($this->getWorld()->getCollidingEntities($this->getBoundingBox(), $this) as $entity){
+				if(!$entity instanceof Living || isset($this->victims[$entity->getId()])){
+					continue;
+				}
+
+				$entityPosition = $entity->getPosition();
+				$xDiff = $entityPosition->getX() - $this->location->getX();
+				$zDiff = $entityPosition->getZ() - $this->location->getZ();
+				if(($xDiff ** 2 + $zDiff ** 2) > $this->radius ** 2){
+					continue;
+				}
+
+				foreach($this->getCloudEffects() as $effect){
+					if($effect->getType() instanceof InstantEffect){
+						$effect->getType()->applyEffect($entity, $effect, 0.5, $this);
+					}else{
+						$entity->getEffects()->add($effect->setDuration((int) round($effect->getDuration() / 4)));
+					}
+				}
+				if($this->reapplicationDelay !== 0){
+					$this->victims[$entity->getId()] = $this->age + $this->reapplicationDelay;
+				}
+				if($this->radiusOnUse !== 0.0){
+					$radius = $this->radius + $this->radiusOnUse;
+					if($radius <= 0){
+						$this->flagForDespawn();
+						return true;
+					}
+					$this->setRadius($radius);
+				}
+				if($this->durationOnUse !== 0){
+					$duration = $this->duration + $this->durationOnUse;
+					if($duration <= 0){
+						$this->flagForDespawn();
+						return true;
+					}
+					$this->setDuration($duration);
+				}
+			}
+			$this->setWaiting($this->age + self::WAIT_TIME);
+			$hasUpdate = true;
+		}
+
+		return $hasUpdate;
+	}
+
+	/**
+	 * Returns the effects the area effect cloud provides.
+	 *
+	 * Used to get COPIES to avoid accidentally modifying the same effect instance
+	 * already applied to another entity.
+	 *
+	 * @return EffectInstance[]
+	 */
+	public function getCloudEffects() : array{
+		return array_map(function(EffectInstance $effect) : EffectInstance{
+			return clone $effect;
+		}, $this->effectContainer->all());
+	}
+
+	protected function syncNetworkData(EntityMetadataCollection $properties) : void{
+		parent::syncNetworkData($properties);
+
+		$spawnTime = $this->getWorld()->getTime() - $this->age;
+		$properties->setInt(EntityMetadataProperties::AREA_EFFECT_CLOUD_SPAWN_TIME, $spawnTime);
+		$properties->setInt(EntityMetadataProperties::AREA_EFFECT_CLOUD_DURATION, $this->duration);
+		$properties->setFloat(EntityMetadataProperties::AREA_EFFECT_CLOUD_RADIUS, $this->initialRadius);
```

Comment:
Why isn't this using the current radius?

Reply:
The client expects InitialRadius, as I mentioned in a comment the client calculates the radius on its own using SpawnTime and InitialRadius. If we send the current radius it will result in that on the client side the cloud will disappear much earlier than it should. See: https://github.com/df-mc/dragonfly/issues/665

Reply:
See: https://github.com/df-mc/dragonfly/issues/665#issuecomment-1338609858

Reply:
A hack has been implemented which consists of sending incorrect values in the entity properties `AREA_EFFECT_CLOUD_RADIUS_CHANGE_ON_PICKUP` and `AREA_EFFECT_CLOUD_RADIUS_PER_TICK`, to disable client-sided radius calculation.